components/net_traits/image_cache_thread.rs

```diff
@@ -37,6 +39,12 @@ pub enum ImageState {
     NotRequested,
 }
 
+impl From<bincode::serde::DeserializeError> for ImageState {
+    fn from(_: bincode::serde::DeserializeError) -> ImageState {
+        ImageState::LoadError
+    }
+}
+
```

Comment:
We could use [`map_err`](https://doc.rust-lang.org/core/result/enum.Result.html#method.map_err) instead of introducing another dependency for the sake of returning an error :)